src/expr.rs

```diff
@@ -1335,7 +1335,10 @@ fn rewrite_struct_lit<'a>(context: &RewriteContext,
         },
         indent: indent,
         width: budget,
-        ends_with_newline: false,
+        ends_with_newline: match context.config.struct_lit_style {
```

Comment:
I think you should look at `tactic` here, rather than `context.config.struct_lit_style`, since we don't just follow that in terms of whether the struct lit should be multi-line or not.